### PR TITLE
[Merged by Bors] - feat(analysis/specific_limits): Lemma for limit of 1 / n as n → ∞ in real algebras

### PR DESCRIPTION
**Pull request by xgenereux**
This PR introduces a single new lemma about the limit of 1 / n as n → ∞ in the complex numbers. It has been placed in a new file to avoid import creep: the obvious file in which to put it (Analysis.SpecificLimits.Basic) does not have the required imports.
  
Note that this is a prerequisite for an upcoming PR for the Hadamard three-line theorem. Finally, thanks to Patrick Massot for supplying the actual proof on Zulip a while back!

---

From the [PR](https://github.com/leanprover-community/mathlib/pull/18887) on mathlib3:

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecificLimits/IsROrC.lean [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+-/
+
+open Set Algebra Filter
+
+variable (𝕜 : Type _) [IsROrC 𝕜]
+
+theorem IsROrC.tendsto_inverse_atTop_nhds_0_nat : 
+    Tendsto (fun n : ℕ => (n : 𝕜)⁻¹) atTop (nhds 0) := by
+  rw [show (fun n : ℕ => (n : 𝕜)⁻¹) = (↑) ∘ fun n : ℕ => (n : ℝ)⁻¹ by ext1 n; simp]
+  exact tendsto_algebraMap_inverse_atTop_nhds_0_nat 𝕜
```
Comment (j-loreaux) [inline Mathlib/Analysis/SpecificLimits/IsROrC.lean:20]:
```suggestion
  convert tendsto_algebraMap_inverse_atTop_nhds_0_nat 𝕜
  simp
```

File: Mathlib/Analysis/SpecificLimits/IsROrC.lean [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/-
+Copyright (c) 2023 Xavier Généreux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Généreux, Patrick Massot
+-/
+import Mathlib.Analysis.SpecificLimits.Basic
+import Mathlib.Analysis.Complex.ReImTopology
+
+/-!
+# A collection of specific limit computations for is_R_or_C
```
Comment (j-loreaux) [inline Mathlib/Analysis/SpecificLimits/IsROrC.lean:10]:
```suggestion
# A collection of specific limit computations for `IsROrC`

```

**Review by j-loreaux — COMMENTED** [review]
I've requested some small changes. Please update the PR title to be more informative about the things you have added.

File: Mathlib/Analysis/SpecificLimits/Basic.lean [hunk 3]
```diff
@@ -53,6 +53,12 @@ theorem tendsto_one_div_add_atTop_nhds_0_nat :
   (tendsto_add_atTop_iff_nat 1).2 (_root_.tendsto_const_div_atTop_nhds_0_nat 1)
 #align tendsto_one_div_add_at_top_nhds_0_nat tendsto_one_div_add_atTop_nhds_0_nat
 
+theorem tendsto_algebraMap_inverse_atTop_nhds_0_nat (𝕜 : Type _) [Semiring 𝕜] [Algebra ℝ 𝕜]
+    [TopologicalSpace 𝕜] [TopologicalSemiring 𝕜] [ContinuousSMul ℝ 𝕜] :
```
[the remark this answers is not in the document]
Comment (j-loreaux) [reply Mathlib/Analysis/SpecificLimits/Basic.lean:57]: For instance, this works:
```
theorem NNReal.tendsto_algebraMap_inverse_atTop_nhds_0_nat (𝕜 : Type _) [Semiring 𝕜] [Algebra ℝ≥0 𝕜]
    [TopologicalSpace 𝕜] [TopologicalSemiring 𝕜] [ContinuousSMul ℝ≥0 𝕜] :
    Tendsto (algebraMap ℝ≥0 𝕜 ∘ fun n : ℕ => (n : ℝ≥0)⁻¹) atTop (nhds 0) := by
  convert (continuous_algebraMap ℝ≥0 𝕜).continuousAt.tendsto.comp tendsto_inverse_atTop_nhds_0_nat
  rw [map_zero (algebraMap ℝ≥0 𝕜)]

theorem tendsto_algebraMap_inverse_atTop_nhds_0_nat (𝕜 : Type _) [Semiring 𝕜] [Algebra ℝ 𝕜]
    [TopologicalSpace 𝕜] [TopologicalSemiring 𝕜] [ContinuousSMul ℝ 𝕜] :
    Tendsto (algebraMap ℝ 𝕜 ∘ fun n : ℕ => (n : ℝ)⁻¹) atTop (nhds 0) :=
  NNReal.tendsto_algebraMap_inverse_atTop_nhds_0_nat 𝕜
```

**Comment by Louddy** [comment]
Thank you for your suggestions @j-loreaux!

**Comment by j-loreaux** [comment]
Thanks!

bors r+

**Comment by j-loreaux** [comment]
Let's try this again...

bors r+

**Pull request closed without merging**